### ISSUE TITLE
Delete origin/branch doesn't work

### ISSUE THREAD

**Issue by firedev** [rowanj/gitx#309]
It removed the branch from the list of origin branches in GitX, but the branches are never removed on origin. If you fetch origin – all the branches are back.

**Comment by rowanj** [comment]
Yep, that's why I disabled it (see #317) in the 0.15 stream.  It only deletes the local copy of the remote ref (what you saw the last time you fetched/pulled from that remote); to delete a branch from a remote is not currently possible within GitX.

So, strangely I see this as both a bug, and a missing feature.

**Comment by firedev** [comment]
`git push origin :branch` would do I think.

**Comment by rowanj** [comment]
Yep.  Just need to make the interface "right" and still have the capability to remove just the ref.

e.g.: if the remote branch has been removed, the push will fail; and the local copy of the ref wouldn't be deleted either.

Fetching from a remote where a branch has been removed also doesn't delete the local copies of the now-missing refs unless '--prune' is specified, and specifying '--prune' by default violates an important git safety net against data loss.

Since remotes are by their nature typically shared by more than one user, the convenience of deletion of the remote branch (once) has to let the case of updates (_n_ collaborators) take precedence until both cases can be handled cleanly.

It's not far away, it's just not quite ready for the quick fix.

**Comment by firedev** [comment]
Wow, sorry, I couldn't even think this far. I always do `git fetch -p` so I got used to the fact that if the branch is removed from origin, I don't need it too.

**Comment by rowanj** [comment]
No problem at all!

The point is that doesn't really do what pretty much anybody wants or expects - mostly I'm just expanding on these issues to help it stick in my mind; but I'm also thinking it might be useful for me to expand on some of these issues that need to be considered, in order to inspire (or intimidate, I guess) people looking to help out.
